demos/pedestrian_tracker_demo/src/cnn.cpp

```diff
@@ -77,7 +77,9 @@ void CnnBase::InferBatch(
             matU8ToBlob<uint8_t>(frames[batch_i + b], input_blob_, b);
         }
 
-        infer_request_.SetBatch(current_batch_size);
+        if ((deviceName_.find("MYRIAD") == std::string::npos) && (deviceName_.find("HDDL") == std::string::npos)) {
+             infer_request_.SetBatch(current_batch_size);    // std::cout << "No dynamic batch support for " << deviceName_ << ". Falling back to batch 1" << std::endl << std::endl;
```

Comment:
I suggest removing that `// std::cout << "No dynamic batch support for " << deviceName_ << ". Falling back to batch 1" << std::endl << std::endl;`: put it after `reid_config.max_batch_size = 1;` and slightly clarify the message:

`std::cerr << "Dynamic batch is not supported for " << deviceName_ << ". Fall back to batch 1\n";` no second std::endl

Reply:
The issue is that it will print many times and could pollute log.

Reply:
Used lambda to print once.

Reply:
Why don't you want to put this print after `reid_config.max_batch_size = 1;` as I said in the first comment? That would eliminate the need in your trick and warn a user in the beginning, but not at the moment when reid is actually triggered (which can happen at a random moment of the video).

Also it seems there is a clearer way to check in `main()` if dynamic batch supported by checking `ie.GetConfig(device, CONFIG_KEY(DYN_BATCH_ENABLED))`. https://github.com/opencv/open_model_zoo/blob/9ef91c63f968208b7a85caaed39249c93ae3050b/demos/smart_classroom_demo/main.cpp#L424

And as for `CnnDLSDKBase::InferBatch`'s check for device, it probably can be replaced by just a check that `input` blob `batch_size` matches `current_batch_size`. And if not, call `infer_request_.SetBatch(current_batch_size);`.

Reply:
Thanks for cleaner way (future proof way when VPU started supporting Batching). I'm not very clear on second point.I need the condition just because calling SetBatch throws exception.

Reply:
`Core::GetConfig` can throw an exception so you need not only check the value, but also catch `InferenceEngine::details::InferenceEngineException`.

As for the second point, I realized now, it won't work. There is slightly another solution, see my suggestion.